crates/dbsp/src/profile/mod.rs

```diff
@@ -174,6 +174,14 @@ impl WorkerProfile {
     pub fn total_shared_bytes(&self) -> Result<HumanBytes, MetaItem> {
         self.attribute_total_as_bytes(SHARED_BYTES_LABEL)
     }
+
+    pub fn merge(&mut self, other: &Self) {
+        for (id, dst) in self.metadata.iter_mut() {
```

Comment:
It probably doesn't happen now, but if `other` has a different set of metrics, some of them can get lost